src/app/core/demo-data/demo-data-initializer.service.ts

```diff
@@ -69,7 +69,12 @@ export class DemoDataInitializerService {
       DemoDataGeneratingProgressDialogComponent,
     );
 
-    await this.demoDataService.publishDemoData();
+    if (window["ndbDemoData"]) {
+      const db = this.dbResolver.getDatabase();
+      await db.putAll(window["ndbDemoData"]);
```

Comment:
if we are saving the demo data within the Angular app, couldn't we do the `entities.map((e) => entitySchemaService.transformEntityToDatabaseFormat(e));` only here and save the hassle to init that in the e2e fixture?

I know we talked about why this didn't work - but I forgot or am not sure if those problems still apply in the final architecture here now :blush:

Reply:
Unfortunately this doesn’t work since `transformEntityToDatabaseFormat(entity)` expects entity to be an instance of `Entity` so that `entity.getSchema()` is defined. But `window.demoData` is only raw JSON. We can only send JSON from the the test runner to the browser.